### ISSUE TITLE
SQLDataType.null generated when using XMLDatabase with lower case array type

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#12611]
An integration test fails when we change

```xml
<data_type>INTEGER ARRAY</data_type>
```

To:

```xml
<data_type>integer array</data_type>
```

The generated code should be:

```java
public final TableField<TRecord, Integer[]> I = 
    createField(DSL.name("I"), SQLDataType.INTEGER.getArrayDataType(), this, "");
```

But is:

```java
public final TableField<TRecord, Integer[]> I = 
    createField(DSL.name("I"), SQLDataType.null, this, "");
```

There shouldn't be any requirement of case sensitive naming of these data types.

**Comment by lukaseder** [comment]
Related: https://github.com/jOOQ/jOOQ/issues/9067

**Comment by lukaseder** [comment]
Fixed in jOOQ 3.16.0, 3.15.5 (https://github.com/jOOQ/jOOQ/issues/12613), and 3.14.16 (https://github.com/jOOQ/jOOQ/issues/12614)
